extensions/resource-deployment/src/services/platformService.ts

```diff
@@ -2,35 +2,62 @@
  *  Copyright (c) Microsoft Corporation. All rights reserved.
  *  Licensed under the Source EULA. See License.txt in the project root for license information.
  *--------------------------------------------------------------------------------------------*/
-
+import * as azdata from 'azdata';
 import * as fs from 'fs';
+import * as cp from 'promisify-child-process';
+import * as sudo from 'sudo-prompt';
 import * as vscode from 'vscode';
-import * as azdata from 'azdata';
-import * as cp from 'child_process';
+import * as nls from 'vscode-nls';
+import { OsType } from '../interfaces';
+
+
+const localize = nls.loadMessageBundle();
+const extensionOutputChannel = localize('resourceDeployment.outputChannel', 'Deployments');
+const sudoPromptTitle = 'AzureDataStudio';
 
 /**
  * Abstract of platform dependencies
  */
 export interface IPlatformService {
+	osType(): OsType;
 	platform(): string;
 	storagePath(): string;
 	copyFile(source: string, target: string): Promise<void>;
 	fileExists(file: string): Promise<boolean>;
 	openFile(filePath: string): void;
-	showErrorMessage(message: string): void;
+	getErrorMessage(error: any): string;
+	showErrorMessage(error: Error | string): void;
+	logToOutputChannel(data: string | Buffer, header?: string): void;
+	outputChannelName(): string;
+	showOutputChannel(preserveFocus?: boolean): void;
 	isNotebookNameUsed(title: string): boolean;
 	makeDirectory(path: string): Promise<void>;
+	ensureDirectoryExists(directory: string): Promise<void>;
 	readTextFile(filePath: string): Promise<string>;
-	runCommand(command: string, options?: CommandOptions): Promise<string>;
+	runCommand(command: string, options?: CommandOptions, sudo?: boolean, commandTitle?: string, ignoreError?: boolean): Promise<string>;
```

Comment:
add these as options to the CommandOptions interface

Reply:
done